Dockerfile

```diff
@@ -21,3 +22,6 @@ EXPOSE 53/udp
 ENTRYPOINT ["docker-entrypoint.sh"]
 
 CMD ["dnsmasq"]
+
+# Health check script
+HEALTHCHECK --interval=5s --timeout=1s --retries=12 CMD ["/opt/healthcheck.sh"]
```

Comment:
I guess interval could be 2s

Reply:
I lowered retries instead. Container now has 15s (5x3) to start, which should be sufficient.